### PR TITLE
BECS support

### PR DESCRIPTION
**Pull request by joannasese**
Completed:

- Added `bsb_code` field
- Included unit tests

Affects the following endpoints:
- `PUT /v2/accounts/{account_id}/billing_info`
- `PUT /v2/accounts/{account_id}`
- `POST /v2/accounts`
- `POST /v2/purchases`
- `POST /v2/purchases/preview`
- `POST /v2/subscriptions`

To create billing_info with BECS:
```ruby
account.billing_info = {
    address1: '123 Fake Boulevard',
    city: 'Adelaide',
    country: 'AU',
    type: 'becs',
    bsb_code: '082-082',
    account_number: '12345678',
    name_on_account: "Chickpea Lentil"
}

account.billing_info.save!
```

### CODE REVIEW

File: lib/recurly/billing_info.rb [hunk 1]
```diff
@@ -39,36 +40,39 @@ class BillingInfo < Resource
...
         attributes |= CREDIT_CARD_ATTRIBUTES
       when 'paypal'
-        attributes -= (CREDIT_CARD_ATTRIBUTES | BANK_ACCOUNT_ATTRIBUTES + AMAZON_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES)
+        attributes -= (CREDIT_CARD_ATTRIBUTES | BANK_ACCOUNT_ATTRIBUTES + AMAZON_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES + BECS_ATTRIBUTES)
       when 'amazon'
-        attributes -= (CREDIT_CARD_ATTRIBUTES | BANK_ACCOUNT_ATTRIBUTES + PAYPAL_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES)
+        attributes -= (CREDIT_CARD_ATTRIBUTES | BANK_ACCOUNT_ATTRIBUTES + PAYPAL_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES + BECS_ATTRIBUTES)
       when 'bank_account'
-        attributes -= (CREDIT_CARD_ATTRIBUTES + PAYPAL_ATTRIBUTES + AMAZON_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES)
+        attributes -= (CREDIT_CARD_ATTRIBUTES + PAYPAL_ATTRIBUTES + AMAZON_ATTRIBUTES + ROKU_ATTRIBUTES + SEPA_ATTRIBUTES + BACS_ATTRIBUTES + BECS_ATTRIBUTES)
```
Comment (bhelx) [inline lib/recurly/billing_info.rb:57]: We really need to re-write this whole method I think. Getting hard to follow what it's doing.

Comment (bhelx) [reply lib/recurly/billing_info.rb:57]: Don't need to do it in this PR though. it doesn't have any tests.

Comment (joannasese) [reply lib/recurly/billing_info.rb:57]: Heard that. I went ahead and made an internal ticket for this.
